internal/envoy/v3/bootstrap.go

```diff
@@ -46,7 +46,7 @@ func WriteBootstrap(c *envoy.BootstrapConfig) error {
 	}
 
 	if c.ResourcesDir != "" {
-		if err := os.MkdirAll(path.Join(c.ResourcesDir, "sds"), 0750); err != nil {
+		if err := os.MkdirAll(path.Join(c.ResourcesDir, "sds"), 0777); err != nil {
```

Comment:
I'd like a comment here linking back to the issue and explaining why we are doing this, having any directory containing secrets as `0777` is very unusual, we need to explain that in this case it's okay because these are actually pointers to secrets that are mounted in via Secret volume mounts (which have their own separate set of controls).

That way, when this is queried in a future security audit, we'll all be able to remember why we did it. :)

Reply:
Pushed fix. PTAL.